inc/saiswitch.h

```diff
@@ -400,6 +400,108 @@ typedef enum _sai_tunnel_decap_ecn_mode_t
 
 } sai_tunnel_decap_ecn_mode_t;
 
+/**
+ * @brief SAI switch soft error recovery error type.
+ */
+typedef enum _sai_switch_soft_error_type_t
+{
+    /**
+     * @brief Unknown error type
+     */
+    SAI_SWITCH_SOFT_ERROR_TYPE_UNKNOWN = 0,
+
+    /**
+     * @brief Parity error
+     */
+    SAI_SWITCH_SOFT_ERROR_TYPE_PARITY = 1,
+
+    /**
+     * @brief ECC single bit error
+     */
+    SAI_SWITCH_SOFT_ERROR_TYPE_ECC_SINGLE_BIT = 2,
+
+    /**
+     * @brief ECC double bit error
+     */
+    SAI_SWITCH_SOFT_ERROR_TYPE_ECC_DOUBLE_BIT = 3,
+
+    /**
+     * @brief All soft error types
+     */
+    SAI_SWITCH_SOFT_ERROR_TYPE_ALL = 4
+} sai_switch_soft_error_type_t;
+
+/**
+ * @brief SAI switch soft error recovery error correction type.
+ */
+typedef enum _sai_switch_correction_type_t
+{
+    /**
+     * @brief S/W takes no action when error
+     * happens (Like some working memories)
+     */
+    SAI_SWITCH_CORRECTION_TYPE_NO_ACTION = 0,
+
+    /**
+     * @brief S/W tries to correct error but fails
+     */
+    SAI_SWITCH_CORRECTION_TYPE_FAIL_TO_CORRECT = 1,
+
+    /**
+     * @brief S/W writes NULL entry to clear the error
+     */
+    SAI_SWITCH_CORRECTION_TYPE_ENTRY_CLEAR = 2,
+
+    /**
+     * @brief Restore entry from a valid S/W cache
+     */
+    SAI_SWITCH_CORRECTION_TYPE_CACHE_RESTORE = 3,
+
+    /**
+     * @brief Restore entry from another pipe
+     */
+    SAI_SWITCH_CORRECTION_TYPE_HW_CACHE_RESTORE = 4,
+
+    /**
+     * @brief Memory needs special correction handling
+     */
+    SAI_SWITCH_CORRECTION_TYPE_SPECIAL = 5,
+} sai_switch_correction_type_t;
+
+/* soft error recovery log info flags */
+#define SAI_SWITCH_SER_LOG_MEM          0x00000001 /* Error happens on memory */
+#define SAI_SWITCH_SER_LOG_REG          0x00000002 /* Error happens on register */
+#define SAI_SWITCH_SER_LOG_MULTI        0x00000004 /* Parity errors detected more than once */
+#define SAI_SWITCH_SER_LOG_CORRECTED    0x00000008 /* Error be corrected by S/W */
+#define SAI_SWITCH_SER_LOG_ENTRY        0x00000010 /* Corrupt memory entry data is read in logfile */
+#define SAI_SWITCH_SER_LOG_CACHE        0x00000020 /* Cache data is valid */
```

Comment:
this seems to be flags, why this is not enum ?

Reply:
take a look at sai_stats_mode_t

Reply:
> take a look at sai_stats_mode_t

Good suggestion. Committed the review suggestion.

https://github.com/opencomputeproject/SAI/pull/1307/commits/ddf25999d43ec08e779565759450a3694e52e8d2